### ISSUE TITLE
Feature Request: DeepStack URL per Camera option

### ISSUE THREAD

**Issue by InTheAir17** [VorlonCD/bi-aidetection#88]
Hi Again,

I have a license plate (LPR) camera and I'm using it with DeepStack. My issue is that the rest of my cameras use DeepStack in High mode (avg. 300ms response) but I use Low mode (avg. 100ms response) for my LPR camera so I need two DeepStack servers. It would be nice to be able to assign a camera to a specific DeepStack instance instead of maintaining/running two version of your application. It works but I see they are sharing the Windows registry entries, so that probably isn't optimal.

Thanks.

**Comment by Tinbum1** [comment]
With the new Amazon Rekognition ability this could also help with identification at night. Combined with a day / night option images could be sent per camera to Amazon Rekognition only at night to utilize the free 5000 images. Or perhaps images could only be sent to Amazon Rekognition if deepstack hasn't already identified anything.

**Comment by Tinbum1** [comment]
Wow, thanks Chris.

**Comment by VorlonCD** [comment]
The latest version has a lot more per-url control in that respect:
[AITOOL-VORLONCD.zip](https://github.com/VorlonCD/bi-aidetection/files/5737452/AITOOL-VORLONCD.zip)

**Comment by InTheAir17** [comment]
Something is wrong, I get this error regarding TEST_CAM in my log for all cameras and nothing shows up in the history tab.

Thanks Paul.

`

Date | Func | Detail | Level | Source | AIServer | Camera | Image | Idx | Depth | Color | ThreadID | FromFile | Filename
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
12/24/2020 12:39:13 AM | AddImageToQueue | ====================== Adding new image to queue (Count=1): E:\BlueIris\AI_Images_LPR\ZZ-AI.20201224_003913216.jpg | Debug | AITOOLS.EXE | None | ZZ-AI | ZZ-AI.20201224_003913216.jpg | 96 | 1 |   | 13 | False | aitool.[2020-12-24].log
12/24/2020 12:39:13 AM | ImageQueueLoop | Adding task for file 'ZZ-AI.20201224_003913216.jpg' (Image QueueTime='0.0' mins, URL Queue wait='6ms', URLOrder=2, URLOriginalOrder=2) on URL 'http://192.168.1.7:81/v1/vision/detection' | Debug | AITOOLS.EXE | 192.168.1.7:81 | ZZ-AI | ZZ-AI.20201224_003913216.jpg | 97 | 0 |   | 8 | False | aitool.[2020-12-24].log
12/24/2020 12:39:13 AM | ImageQueueLoop | Done adding 1 total threads, ErrCnt=0, ImageProcessQueue.Count=0 | Debug | AITOOLS.EXE | 192.168.1.7:81 | ZZ-AI | None | 98 | 0 |   | 8 | False | aitool.[2020-12-24].log
12/24/2020 12:39:13 AM | DetectObjects | Starting analysis of E:\BlueIris\AI_Images_LPR\ZZ-AI.20201224_003913216.jpg... | Debug | AITOOLS.EXE | 192.168.1.7:81 | ZZ-AI | ZZ-AI.20201224_003913216.jpg | 99 | 1 |   | 11 | False | aitool.[2020-12-24].log
12/24/2020 12:39:13 AM | IsValidImage | Image file is valid: ZZ-AI.20201224_003913216.jpg | Debug | AITOOLS.EXE | 192.168.1.7:81 | ZZ-AI | None | 100 | 2 |   | 9 | False | aitool.[2020-12-24].log
12/24/2020 12:39:13 AM | GetDetectionsFromAIServer | (1/6) Uploading a 8260 byte image to 'DeepStack' AI Server at http://192.168.1.7:81/v1/vision/detection | Debug | AITOOLS.EXE | 192.168.1.7:81 | ZZ-AI | ZZ-AI.20201224_003913216.jpg | 101 | 1 |   | 9 | False | aitool.[2020-12-24].log
12/24/2020 12:39:13 AM | DetectObjects | (2/6) Posted in 287ms, StatusCode='OK', Received a 33 byte JSON response: '{"success":true,"predictions":[]...' | Debug | AITOOLS.EXE | 192.168.1.7:81 | ZZ-AI | ZZ-AI.20201224_003913216.jpg | 102 | 1 |   | 13 | False | aitool.[2020-12-24].log
12/24/2020 12:39:13 AM | DetectObjects | (3/6) Processing results... | Debug | AITOOLS.EXE | 192.168.1.7:81 | ZZ-AI | ZZ-AI.20201224_003913216.jpg | 103 | 1 |   | 13 | False | aitool.[2020-12-24].log
12/24/2020 12:39:13 AM | DetectObjects | ((NO DETECTED OBJECTS)) | Debug | AITOOLS.EXE | 192.168.1.7:81 | ZZ-AI | ZZ-AI.20201224_003913216.jpg | 104 | 1 |   | 13 | False | aitool.[2020-12-24].log
12/24/2020 12:39:13 AM | DetectObjects | (5/6) Performing CANCEL actions: | Debug | AITOOLS.EXE | 192.168.1.7:81 | ZZ-AI | ZZ-AI.20201224_003913216.jpg | 105 | 3 |   | 13 | False | aitool.[2020-12-24].log
12/24/2020 12:39:13 AM | DetectObjects | (6/6) Camera ZZ-AI caused a false alert, nothing detected. | Debug | AITOOLS.EXE | 192.168.1.7:81 | ZZ-AI | ZZ-AI.20201224_003913216.jpg | 106 | 2 |   | 13 | False | aitool.[2020-12-24].log
12/24/2020 12:39:13 AM | SaveAsync | JSON Settings saved to REGISTRY and C:\AITool\_Settings\aitool.Settings.JSON | Debug | AITOOLS.EXE | 192.168.1.7:81 | ZZ-AI | None | 107 | 2 |   | 11 | False | aitool.[2020-12-24].log
12/24/2020 12:39:13 AM | DetectObjects | Total Time:  438ms (Count=1, Min=438ms, Max=438ms, Avg=438ms) | Debug | AITOOLS.EXE | 192.168.1.7:81 | ZZ-AI | ZZ-AI.20201224_003913216.jpg | 108 | 2 |   | 13 | False | aitool.[2020-12-24].log
12/24/2020 12:39:13 AM | DetectObjects | AI (URL) Time:  287ms (Count=1, Min=287ms, Max=323.0ms, Avg=287ms) | Debug | AITOOLS.EXE | 192.168.1.7:81 | ZZ-AI | ZZ-AI.20201224_003913216.jpg | 109 | 2 |   | 13 | False | aitool.[2020-12-24].log
12/24/2020 12:39:13 AM | DetectObjects | File lock Time:  5ms (Count=1, Min=5ms, Max=5ms, Avg=5ms) | Debug | AITOOLS.EXE | 192.168.1.7:81 | ZZ-AI | ZZ-AI.20201224_003913216.jpg | 110 | 2 |   | 13 | False | aitool.[2020-12-24].log
12/24/2020 12:39:13 AM | DetectObjects | Image Queue Time:  91ms (Count=1, Min=91ms, Max=91ms, Avg=91ms) | Debug | AITOOLS.EXE | 192.168.1.7:81 | ZZ-AI | ZZ-AI.20201224_003913216.jpg | 111 | 2 |   | 13 | False | aitool.[2020-12-24].log
12/24/2020 12:39:13 AM | DetectObjects | Image Queue Depth:  1 (Count=1, Min=1, Max=1, Avg=1) | Debug | AITOOLS.EXE | 192.168.1.7:81 | ZZ-AI | ZZ-AI.20201224_003913216.jpg | 112 | 2 |   | 13 | False | aitool.[2020-12-24].log
12/24/2020 12:39:22 AM | GetCamera | No enabled camera with the same filename, cameraname, or prefix found for 'TEST_CAM' | Debug | AITOOLS.EXE | 192.168.1.7:81 | ZZ-AI | None | 113 | 0 |   | 1 | False | aitool.[2020-12-24].log
12/24/2020 12:39:22 AM | GetCamera | No default camera found. Aborting. | Warn | AITOOLS.EXE | 192.168.1.7:81 | ZZ-AI | None | 114 | 0 |   | 1 | False | aitool.[2020-12-24].log
12/24/2020 12:39:22 AM | GetCamera | **Cannot match 'TEST_CAM' to an existing camera.** | Debug | AITOOLS.EXE | 192.168.1.7:81 | ZZ-AI | None | 115 | 0 |   | 1 | False | aitool.[2020-12-24].log
12/24/2020 12:39:22 AM | GetCamera | **No enabled camera with the same filename, cameraname, or prefix found for 'TEST_CAM'** | Debug | AITOOLS.EXE | 192.168.1.7:81 | ZZ-AI | None | 116 | 0 |   | 1 | False | aitool.[2020-12-24].log
12/24/2020 12:39:22 AM | GetCamera | No default camera found. Aborting. | Warn | AITOOLS.EXE | 192.168.1.7:81 | ZZ-AI | None | 117 | 0 |   | 1 | False | aitool.[2020-12-24].log
12/24/2020 12:39:22 AM | GetCamera | **Cannot match 'TEST_CAM' to an existing camera.** | Debug | AITOOLS.EXE | 192.168.1.7:81 | ZZ-AI | None | 118 | 0 |   | 1 | False | aitool.[2020-12-24].log

`

**Comment by InTheAir17** [comment]
My bad, I had a filter on in the history, though I'm not sure why I'm seeing that TEST_CAM. 

I think I figured out the other thing as well. I created a second AI server for LPR and assigned the one camera that uses LPR to it. It looks like the AI server list uses rules so I moved that server to the top of the list so the second one becomes my default for all other cameras... i.e. I don't have to do anything else for the other cameras, they just work.

Thanks for this excellent update.
